### DISCUSSION TITLE
Dose Influence Matrix for Each Structure

### DISCUSSION THREAD

**Discussion by BaharM1997** [e0404/matRad#589] [Q&A] [answered]
Hi,

I want to build my own optimization model, but use the data from matRad. I have imported DICOM data into matRad and calculated the dose influence matrix. I know that this matrix is stored in dij.physicalDose , but I need the dose influence matrix for each structure separately. So my question is how can I know which rows of the dose influence matrix belong to which structure?

Thank you in advance for your time!

**Comment by wahln** [comment, marked as answer]
The structure indices are stored in the 4th column of the `cst` for each structure. For example, `cst{i,4}{j}` is the index list of the i-th structure, for the j-th ct scenario (usually,` j=1`). You can use this to index the dose influence matrix then: `dij.physicalDose{1}(cst{1,4}{1},:)`
Note that structures may overlap.

**Reply by BaharM1997** [reply]
Thank you for your reply!

I have tried this before but I got the error "Index exceeds array bounds." since for instance the indices of my target starts at 10062017 but the dimension of my dose influence matrix is 3067790 * 83359.

**Reply by wahln** [reply]
Sorry, I forgot that the dose grid and c(s)t grids can be different. During dose calculation (and optimization), your cst get's temporarily resampled to the defined dose grid.
If you want the same resolution as the ct, you can set `pln.propDoseCalc.doseGrid.resolution = ct.resolution;` before dose calculation, then the indices are valid inherently.
Alternatively, you can get the downsampled cst and use the downsampled indices:
`cstDownsampled = matRad_resizeCstToGrid(cst,dij.ctGrid.x,dij.ctGrid.y,dij.ctGrid.z,dij.doseGrid.x,dij.doseGrid.y,dij.doseGrid.z);
dij.physicalDose{1}(cst{1,4}{1},:);`

**Comment by BaharM1997** [comment]
Thank you very much for your help! This is very helpful!
